### PR TITLE
[doc]: add 3.7

### PR DESCRIPTION
**Pull request by seognil**
finish #267 

I made some serious effort (maybe 10 hours ?) for understanding and words using
thanks to this by the way
https://jkchao.github.io/typescript-book-chinese/new/typescript-3.7.html
(But I thought some parts are bit weird or maybe even wrong, could be improved)

Would you guys like to make a review, in case of misunderstanding
@zhongsp (as the translator of this repo)
@jkchao (as the translator of TypeScript Deep Dive)

### CODE REVIEW

**Comment by zhongsp** [comment]
@seognil Thanks! I need some time to learn. Can you rebase the code on to master branch? I see some small merging conflicts exist.

**Comment by seognil** [comment]
@zhongsp done

File: doc/release-notes/TypeScript 3.7.md [hunk 1]
```diff
@@ -0,0 +1,848 @@
+# TypeScript 3.7
+
+## 可选链（Optional Chining）
```
Comment (zhongsp) [inline doc/release-notes/TypeScript 3.7.md]: Typo: should be `Chaining`

Comment (seognil) [reply doc/release-notes/TypeScript 3.7.md]: fixed (and some other stuff)
but I saw link problems in `SUMMARY.md` `preface.md` `README.md` while checking links

Should I make a quick fix for these three files just in this PR or later

Comment (zhongsp) [reply doc/release-notes/TypeScript 3.7.md]: Let's merge the main content first. The links are not big problemt. :D

**Pull request merged**